File: ios/ArxOSMobile/ArxOSMobile/Models/UserProfile.swift
//
//  UserProfile.swift
//  ArxOSMobile
//
//  User profile model for storing user credentials and preferences
//

import Foundation

/// User profile for storing local credentials and preferences
struct UserProfile: Codable {
    var name: String
    var email: String
    var company: String?
    
    init(name: String, email: String, company: String? = nil) {
        self.name = name
        self.email = email
        self.company = company
    }
    
    /// Load profile from UserDefaults
    static func load() -> UserProfile? {
        guard let data = UserDefaults.standard.data(forKey: "arxos_user_profile"),
              let profile = try? JSONDecoder().decode(UserProfile.self, from: data) else {
            return nil
        }
        return profile
    }
    
    /// Save profile to UserDefaults
    func save() {
        if let data = try? JSONEncoder().encode(self) {
            UserDefaults.standard.set(data, forKey: "arxos_user_profile")
        }
    }
    
    /// Check if onboarding is needed
    static func needsOnboarding() -> Bool {
        return UserProfile.load() == nil
    }
    
    /// Clear stored profile (for logout/reset)
    static func clear() {
        UserDefaults.standard.removeObject(forKey: "arxos_user_profile")
    }
}

File: android/app/src/main/java/com/arxos/mobile/data/UserProfile.kt
package com.arxos.mobile.data

import android.content.Context
import android.content.SharedPreferences
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class UserProfile(
    val name: String,
    val email: String,
    val company: String? = null
) {
    companion object {
        private const val PREFS_NAME = "arxos_prefs"
        private const val KEY_PROFILE = "user_profile"
        
        /**
         * Load user profile from SharedPreferences
         */
        fun load(context: Context): UserProfile? {
            val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            val json = prefs.getString(KEY_PROFILE, null)
            return if (json != null) {
                try {
                    Json.decodeFromString<UserProfile>(json)
                } catch (e: Exception) {
                    null
                }
            } else {
                null
            }
        }
        
        /**
         * Save user profile to SharedPreferences
         */
        fun save(context: Context, profile: UserProfile) {
            val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            val json = Json.encodeToString(profile)
            prefs.edit().putString(KEY_PROFILE, json).apply()
        }
        
        /**
         * Check if onboarding is needed
         */
        fun needsOnboarding(context: Context): Boolean {
            return load(context) == null
        }
        
        /**
         * Clear user profile (for logout/reset)
         */
        fun clear(context: Context) {
            val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
            prefs.edit().remove(KEY_PROFILE).apply()
        }
    }
}

